### PR TITLE
Add websocket: Add docs + update examples for evm and terra packages

### PR DESCRIPTION
**Pull request by ali-bahjati**

### CODE REVIEW

**Review by cctdaniel — DISMISSED** [review]

File: pyth-evm-js/README.md [hunk 1]
```diff
@@ -41,6 +41,18 @@ const priceFeeds = connection.getPythLatestPriceFeeds(priceIds);
 console.log(priceFeeds[0].getCurrentPrice()); // Price { conf: '1234', expo: -8, price: '12345678' }
 console.log(priceFeeds[1].getEmaPrice()); // Exponentially-weighted moving average price
 
+// By subscribing to the price feeds you can get their updates realtime.
```
Comment (jayantk) [inline pyth-evm-js/README.md:44]: in real-time

File: pyth-evm-js/README.md [hunk 2]
```diff
@@ -41,6 +41,18 @@ const priceFeeds = connection.getPythLatestPriceFeeds(priceIds);
...
+connection.subscribePriceFeedUpdates(priceIds, (priceFeed) => {
+  console.log("Received a new price feed update!");
+  console.log(priceFeed.getCurrentPrice());
+});
+
+// When using subscription, make sure to close the websocket upon termination to finish the process gracefully.
+// connection.closeWebSocket();
```
Comment (jayantk) [inline pyth-evm-js/README.md:51]: delete this commented-out code line?

File: pyth-evm-js/src/examples/EvmPriceServiceClient.ts [hunk 3]
```diff
@@ -22,20 +26,38 @@ const argv = yargs(hideBin(process.argv))
...
+      `Current price for ${priceFeed.id}: ${JSON.stringify(
+        priceFeed.getCurrentPrice()
+      )}.`
+    );
+  });
+
+  await sleep(600000);
+
+  // To close the websocket you should either unsubscribe from all
+  // price feeds or call `connection.stopWebSocket()` directly.
```
Comment (jayantk) [inline pyth-evm-js/src/examples/EvmPriceServiceClient.ts:55]: i think it's `closeWebSocket()` not `stopWebSocket()`

File: pyth-evm-js/src/examples/EvmPriceServiceClient.ts [hunk 4]
```diff
@@ -22,20 +26,38 @@ const argv = yargs(hideBin(process.argv))
...
+
+  console.log("Unsubscribing from price feed updates.");
+  await connection.unsubscribePriceFeedUpdates(priceIds);
+
+  // connection.closeWebSocket();
```
Comment (jayantk) [inline pyth-evm-js/src/examples/EvmPriceServiceClient.ts:60]: i don't think you need this here (given that it's already mentioned in the earlier comment)

File: pyth-terra-js/src/examples/TerraPriceServiceClient.ts [hunk 5]
```diff
@@ -22,22 +26,42 @@ const argv = yargs(hideBin(process.argv))
   .parseSync();
 
 async function run() {
-  const connection = new TerraPriceServiceConnection({
-    httpEndpoint: argv.http,
+  const connection = new TerraPriceServiceConnection(argv.endpoint, {
+    logger: console, // Providing logger will allow the connection to log it's events.
```
Comment (jayantk) [inline pyth-terra-js/src/examples/TerraPriceServiceClient.ts:30]: this should be "its", not "it's". It's -> it is

File: pyth-evm-js/README.md [hunk 6]
```diff
@@ -83,10 +95,10 @@ There are two examples in [examples](./src/examples/).
 
 #### EvmPriceServiceClient
 
-[This example](./src/examples/EvmPriceServiceClient.ts) fetches a `PriceFeed` for each given price id and prints them. You can run it with `npm run example-client`. A full command that prints BTC and ETH Price Feeds, in the testnet network, looks like so:
+[This example](./src/examples/EvmPriceServiceClient.ts) fetches a `PriceFeed` for each given price id and prints them. It also subscribes for their updates and prints the updates as they come. You can run it with `npm run example-client`. A full command that prints BTC and ETH Price Feeds, in the testnet network, looks like so:
```
Comment (jayantk) [inline pyth-evm-js/README.md:98]: how about something like "This example fetches `PriceFeed` updates using both a HTTP-request API and a streaming websocket API." At the moment, it's not clear that the first fetch is not a subscription thing.

also, don't capitalize "Price Feeds"

**Review by jayantk — APPROVED** [review]
nice, left you some minor wording comments

File: pyth-evm-js/src/examples/EvmPriceServiceClient.ts (hunk 3 above)
Comment (ali-bahjati) [reply pyth-evm-js/src/examples/EvmPriceServiceClient.ts:55]: Thanks for spotting!

File: pyth-evm-js/src/examples/EvmPriceServiceClient.ts (hunk 4 above)
Comment (ali-bahjati) [reply pyth-evm-js/src/examples/EvmPriceServiceClient.ts:60]: Agreed.

File: pyth-terra-js/src/examples/TerraPriceServiceClient.ts (hunk 5 above)
Comment (ali-bahjati) [reply pyth-terra-js/src/examples/TerraPriceServiceClient.ts:30]: yeah.. thank you

File: pyth-evm-js/README.md (hunk 1 above)
Comment (ali-bahjati) [reply pyth-evm-js/README.md:44]: thanks.

File: pyth-evm-js/README.md (hunk 2 above)
Comment (ali-bahjati) [reply pyth-evm-js/README.md:51]: yeah makes sense.

**Review by jayantk — APPROVED** [review]

**Pull request merged**